src/lib/cache/subCache.cpp

```diff
@@ -1198,33 +1204,48 @@ void subCacheSync(void)
   while (cSubP != NULL)
   {
     CachedSubSaved* cssP = savedSubV[cSubP->subscriptionId];
-    if ((cssP != NULL) && (cssP->lastNotificationTime <= cSubP->lastNotificationTime))
+
+    if (cssP != NULL)
     {
-      // cssP->lastNotificationTime older than what's currently in DB => throw away
-      cssP->lastNotificationTime = 0;
+      if (cssP->lastNotificationTime <= cSubP->lastNotificationTime)
+      {
+        // cssP->lastNotificationTime is older than what's currently in DB => throw away
+        cssP->lastNotificationTime = 0;
+      }
+
+      if (cssP->lastFailure < cSubP->lastFailure)
```

Comment:
'What happend when lastFailure is released?': lastFailure is cache or in DB?
Yes, lastFailure is set to -1 in cache when a notification works.
So, cssP->lastFailure is less than what's read from DB and it is thrown away.

I see no problem here, but, let's discuss it so you can explain what the possible problem is.